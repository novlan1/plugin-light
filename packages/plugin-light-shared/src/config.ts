export const PLATFORM_MAP = {
  MP_WX: 'mp-weixin',
  MP_QQ: 'mp-qq',
  MP_ALIPAY: 'mp-alipay',
  H5: 'h5',
} as const;


export const ALL_PLATFORM = 'ALL';


export const PLATFORMS_MP = [
  PLATFORM_MAP.MP_WX,
  PLATFORM_MAP.MP_QQ,
];


export const PLATFORMS_ALL = [
  PLATFORM_MAP.MP_WX,
  PLATFORM_MAP.MP_QQ,
  PLATFORM_MAP.H5,
];


export const HTML_MAP = {
  MP_WX: '.wxml',
  MP_QQ: '.qml',
  MP_ALIPAY: '.axml',
  MP_JD: '.jxml',
} as const;

export const CSS_MAP = {
  MP_WX: '.wxss',
  MP_QQ: '.qss',
  MP_ALIPAY: '.acss',
  MP_JD: '.jxss',
} as const;


export const CSS_POSTFIX_MAP = Object.keys(CSS_MAP)
  .reduce((acc: any, item) => {
    acc[item] = (CSS_MAP[item as keyof typeof CSS_MAP] as string).slice(1);
    return acc;
  }, {});


export const CDN_MAP = {
  UNI_SIMPLE_ROUTER: 'https://image-1251917893.cos.ap-guangzhou.myqcloud.com/igame/npm/uni-simple-router/uni-simple-router%402.0.8-beta.4-1.js',
};
