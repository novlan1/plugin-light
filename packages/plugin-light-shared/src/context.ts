const DEFAULT_KEYS = [
  'UNI_APP_X',
  'APP',
  'APP_UVUE',
  'APP_NVUE',
  'APP_PLUS',
  'APP_PLUS_NVUE',
  'APP_VUE',
  'APP_ANDROID',
  'APP_IOS',
  'APP_HARMONY',
  'H5',
  'MP',
  'MP_360',
  'MP_ALIPAY',
  'MP_BAIDU',
  'MP_QQ',
  'MP_LARK',
  'MP_TOUTIAO',
  'MP_WEIXIN',
  'MP_KUAISHOU',
  'MP_JD',
  'QUICKAPP_NATIVE',
  'QUICKAPP_WEBVIEW',
  'QUICKAPP_WEBVIEW_HUAWEI',
  'QUICKAPP_WEBVIEW_UNION',
  'VUE2',
  'VUE3',
  'WEB',
] as const;


export const DEFAULT_CONTEXT_OBJECT = DEFAULT_KEYS.reduce((acc, key) => ({
  ...acc,
  [key]: false,
}), {});
