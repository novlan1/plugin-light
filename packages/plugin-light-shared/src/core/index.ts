export * from './cross-game-style';
export * from './gen-version';
export * from './ifdef';
export * from './inject-dynamic-style';
export * from './replace-vue-directive';
export * from './v-lazy';
