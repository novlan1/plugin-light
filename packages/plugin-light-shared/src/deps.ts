import path from 'path';
import { readFileSync } from 't-comm';

export function getDeps(dir: string) {
  const data = readFileSync(path.resolve(dir, 'package.json'), true);

  return Object.keys({
    ...data.dependencies,
  });
}
