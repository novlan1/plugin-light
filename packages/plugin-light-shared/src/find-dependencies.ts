/* eslint-disable @typescript-eslint/no-require-imports */
type IPath = {
  parent: string;
  parentPath: IPath;
};

function getParent(path: IPath, level: number): any {
  if (!level) {
    return path.parent;
  }
  return getParent(path.parentPath, level - 1);
}

export function findDependencies(content: string) {
  const parser = require('@babel/parser');
  const traverse = require('@babel/traverse').default;

  const sourceList: Array<string> = [];
  const ast = parser.parse(content, {
    // 不加这个配置，报错：SyntaxError: 'import' and 'export' may appear only with 'sourceType: "module"'
    sourceType: 'module',
    plugins: ['typescript'],
  });

  traverse(ast, {
    CallExpression(path: any) {
      if (path.node.callee.name === 'require') {
        if (path.node.arguments[0].type === 'StringLiteral') {
          sourceList.push(path.node.arguments[0].value);
        }
        if (
          getParent(path, 0).type === 'ExpressionStatement'
          && getParent(path, 1).type === 'BlockStatement'
          && getParent(path, 2).type === 'ObjectMethod'
          && getParent(path, 3).type === 'ObjectExpression'
          && getParent(path, 4)?.type === 'ObjectProperty'
          && getParent(path, 4)?.key?.name === 'components'
          && getParent(path, 5)?.type === 'ObjectExpression'
          && getParent(path, 6)?.type === 'ExportDefaultDeclaration'
          && path.node.arguments[0].type === 'ArrayExpression') {
          sourceList.push(path.node.arguments[0].elements[0].value);
        }
      }
    },
    ImportDeclaration(path: any) {
      sourceList.push(path.node.source.value);
    },
  });
  return sourceList;
}

