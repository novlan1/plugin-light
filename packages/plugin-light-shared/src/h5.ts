export function checkH5() {
  return process.env.VUE_APP_PLATFORM === 'h5';
}
