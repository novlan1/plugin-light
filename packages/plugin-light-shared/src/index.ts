export * from './core';


export {
  checkBundleAnalyze,
  checkDebugMode,
} from './bundle-analyze';

export {
  PLATFORM_MAP,
  ALL_PLATFORM,
  PLATFORMS_ALL,
  PLATFORMS_MP,
  HTML_MAP,
  CSS_MAP,
  CSS_POSTFIX_MAP,
  CDN_MAP,
} from './config';

export {
  DEFAULT_TRANSPILE_DEPENDENCIES,
  DEFAULT_ADAPTER_DIRS,
  AEGIS_EXTERNAL_SCRIPT_LINK,
  UNI_SIMPLE_ROUTER_SCRIPT_LINK,
  EXTERNAL_LINK_MAP,
} from './config-project';

export {
  ROOT_NAME,
  saveJsonToLog,
  createLogDir,
  normalizePath,
  updateAssetSource,
  removeFirstSlash,
  parseSetDeps,
  getRelativePath,
  getUniCliCache,
} from './helper';


export {
  DEFAULT_CONTEXT_OBJECT,
} from './context';


export {
  getDeps,
} from './deps';


export {
  findDependencies,
} from './find-dependencies';


export {
  checkH5,
} from './h5';


export {
  getLoaderFile,
  getLoaderProdFile,
} from './loader-file';

export {
  saveLoaderLog,
  recordLoaderLog,
} from './loader-log';

export {
  getProjectName,
  getSubProjectName,
} from './project-name';

export {
  updateManifest,
  revertManifest,
} from './replace-manifest';

export {
  getRootDir,
} from './root';


export {
  getSubProjectConfig,
  getSubProjectRoot,
} from './sub-project';
