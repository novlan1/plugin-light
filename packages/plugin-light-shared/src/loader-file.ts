import path from 'path';

const LOADER_PROD = 'loader.prod.js';

export function getLoaderFile(dir = '', isProd = false) {
  if (isProd) {
    return path.resolve(dir, LOADER_PROD);
  }
  return path.resolve(dir, 'loader.js');
}


export function getLoaderProdFile(dir = '') {
  return path.resolve(dir, LOADER_PROD);
}

