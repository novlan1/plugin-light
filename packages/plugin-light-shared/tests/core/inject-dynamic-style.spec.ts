import * as fs from 'fs';
import path from 'path';
import { getStyleList, genInjectContent, getComponentName } from '../../src/';


jest.mock('fs');

const dir = path.resolve(process.cwd(), 'src/local-component/ui/tip-match/tip-match-notice/css');
const dirWithoutComponentName = path.resolve(process.cwd(), 'src/local-component/ui/tip-match/tip-match-notice');


describe('injectDynamicStyleWeb', () => {
  const oldEnv = process.env;

  beforeAll(() => {
    jest.spyOn(fs, 'readdirSync').mockReturnValue(['base.scss', 'dialog.scss', 'icon.scss'] as any);
  });

  beforeEach(() => {
    process.env = { ...oldEnv };
    process.env.UNI_PLATFORM = 'h5';
  });

  afterEach(() => {
    process.env = oldEnv;
    jest.restoreAllMocks();
  });

  it('getStyleList', () => {
    expect(getStyleList(dir)).toEqual(['dialog', 'icon']);
  });

  it('getComponentName', () => {
    expect(getComponentName(dir)).toBe('tip-match-notice');

    const dir2 = path.resolve(process.cwd(), 'src/project/gp/components/dialogs/match-result-dialog/css');
    expect(getComponentName(dir2)).toBe('match-result-dialog');

    expect(getComponentName(dirWithoutComponentName)).toBeFalsy();
  });

  it('genInjectContent', () => {
    expect(genInjectContent({
      styleList: ['dialog', 'icon'],
      componentName: 'tip-match-notice',
      topElement: 'body',
    })).toMatchSnapshot();
  });
});
