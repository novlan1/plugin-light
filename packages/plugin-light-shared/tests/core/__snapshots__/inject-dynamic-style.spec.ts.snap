// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injectDynamicStyleWeb genInjectContent 1`] = `
"
body.tip-match-notice {
  
  &--type-dialog {
    @import './dialog.scss';
  }

  &--type-icon {
    @import './icon.scss';
  }
}
"
`;
